testsuite/tests/th/T11721_TH.hs

```diff
@@ -12,15 +12,16 @@ $(return [])
 
 main :: IO ()
 main = print
-  $(do let rightOrder :: [TyVarBndr] -> Bool
-           rightOrder [KindedTV b _, KindedTV a _]
-             = nameBase b == "b" && nameBase a == "a"
-           rightOrder _ = False
+  $(do let varName :: TyVarBndr -> Maybe String
+           varName (KindedTV v _) = Just (nameBase v)
+           varName _ = Nothing
```

Comment:
Is it right of us to fix things this way?

In other words: is this backwards incompatibility something which will affect reasonable code? The "fix" is quite profound, it makes me feel uneasy.

Reply:
OK, I'll remove it for now.